UnitTests/UnitTestController.cs

```diff
@@ -0,0 +1,119 @@
+﻿using GetandTake.Controller;
+using GetandTake.Models.DTOs.ListDTO;
+using GetandTake.Services.Abstract;
+using Moq;
+
+namespace UnitTests;
+
+public class UnitTestController
+{
+    [Fact]
+    public Task ProductList_ShouldExecuteSuccessullyAndReturnProducts()
+    {
+        //arrange
+        var productService = new Mock<IProductService>();
+        var productList = GetProductsAsync();
+        productService.Setup(product => product.GetAllAsync())
+            .Returns(productList);
+        var productController = new ProductController(productService.Object);
+
+        //act
+        var productResult = productController.ProductList();
+
+        //assert
+        Assert.NotNull(productResult);
+        Assert.Equal(productList.Result.Count, productResult.Result.Count);
+        Assert.Equal(productList.ToString(), productResult.ToString());
+        return Task.CompletedTask;
```

Comment:
To avoid this return (it is absolutely not needed) you can add `async` modifier in method signature. Update it for other cases as well, please